### ISSUE TITLE
regarding DAIYN & LSD

### ISSUE THREAD

**Issue by lrubato** [artberryx/CSD-public#1]
This work looks so interesting! 

Are DAIYN and LSD also included in this codebase? if so how can run it?

**Comment by seohongpark** [comment]
Thank you for your interest in this work. LSD is included in the code: you can turn off the flags related to dual gradient descent (e.g., `--dual_reg=0`) and set `--spectral_normalization=1` & `--dual_dist=l2` to run LSD.

For DIAYN, you can use the following code snippet in `discriminator.py`:
```
                if self.skill_type == 'discrete':
                    self.logits_tf = snn(obs2_focus, [int(self.hidden / 2)] * self.layers + [self.dimz], sn=self.spectral_normalization, is_training=self.is_training)
                    self.sk_tf = tf.nn.softmax_cross_entropy_with_logits(labels=tf.stop_gradient(self.z_tf), logits=self.logits_tf, name='mean')
                    self.sk_r_tf = -1 * self.sk_tf
                else:
                    self.mean_tf = snn(obs2_focus, [int(self.hidden / 2)] * self.layers + [self.dimz], name='mean', sn=self.spectral_normalization, is_training=self.is_training)
                    self.log_std_tf = snn(obs2_focus, [int(self.hidden / 2)] * self.layers + [self.dimz], name='log_std', sn=self.spectral_normalization, is_training=self.is_training)
                    self.clamped_log_std_tf = tf.clip_by_value(self.log_std_tf, -13.8155, 1000)
                    self.clamped_std_tf = tf.exp(self.clamped_log_std_tf)
                    self.sk_tf = tf.math.reduce_sum(input_tensor=(((self.mean_tf - self.z_tf) / self.clamped_std_tf) ** 2) / 2 + self.clamped_log_std_tf + 0.5 * np.log(2 * np.pi), axis=1)
                    self.sk_r_tf = -1 * self.sk_tf
```
